plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -941,7 +948,7 @@ private void finishOptimize(ConnectorSession session, IcebergTableExecuteHandle
             newFiles.add(builder.build());
         }
 
-        if (scannedFiles.isEmpty() && newFiles.isEmpty()) {
+        if (scannedDataFiles.isEmpty() && newFiles.isEmpty()) {
```

Comment:
Should this include the delete files?
```suggestion
        if (scannedDataFiles.isEmpty() && scannedDeleteFiles.isEmpty() && newFiles.isEmpty()) {
```

Reply:
I can add it but I am pretty sure that it is impossible for delete files to be not empty here while data files are empty.